bin/Compress/src/compressor.c

```diff
@@ -101,36 +109,50 @@ int main(int argc, char** argv)
     junk = fread(inROM, tempSize, 1, file);
     fclose(file);
 
-    /* Read archive if it exists*/
+    /* Find the file table and relevant info */
+    tabStart = findTable(inROM);
+    fileTab = (uint32_t*)(inROM + tabStart);
+    getTableEnt(&tab, fileTab, 2);
+    tabSize = tab.endV - tab.startV;
+    tabCount = tabSize / 16;
+
+    /* Read archive if it exists */
     file = fopen("ARCHIVE.bin", "rb");
     if(file != NULL)
     {
-        /* Get number of files */
-        printf("Loading Archive.\n");
-        fflush(stdout);
+        /* The table count is the number of tables in the ROM */
+        /* The file count is the number of files in the archive */
         archive = malloc(sizeof(archive_t));
-        junk = fread(&(archive->fileCount), sizeof(uint32_t), 1, file);
-        
-        /* Allocate space for files and sizes */
-        archive->refSize = malloc(sizeof(uint32_t) * archive->fileCount);
-        archive->srcSize = malloc(sizeof(uint32_t) * archive->fileCount);
-        archive->ref = malloc(sizeof(uint8_t*) * archive->fileCount);
-        archive->src = malloc(sizeof(uint8_t*) * archive->fileCount);
+        archive->tabCount = tabCount;
+        junk = fread(&(archive->numFiles), sizeof(uint32_t), 1, file);
+
+        /* We want an archive file for every table entry */
+        /* Initialize all SRC and REF arrays to NULL */
+        archive->files = calloc(sizeof(archiveFile_t), archive->tabCount);
+
+        printf("Loading Archive with %d files.\n", archive->numFiles);
 
         /* Read in file size and then file data */
-        for(i = 0; i < archive->fileCount; i++)
+        for(i = 0; i < archive->numFiles; ++i)
         {
+            /* Get the index that this file goes into */
+            junk = fread(&j, sizeof(uint32_t), 1, file);
```

Comment:
Since this changes the format of ARCHIVE.bin, all existing archives are invalidated. I'm not exactly sure the full extent of this issue but in the best case the ARCHIVE is basically ignored, in the worst case it may result in data corruption until ARCHIVE.bin is replaced with this new format. Suggesting that error checking be implemented such that if an error is detected when processing the ARCHIVE, erase it/overwrite it

Reply:
I was thinking I implemented that, but it doesn't appear so. Yes, this will cause some really sketch behavior if you use an old archive file with the new compressor. My main concern with a version number is that it would be potentially indistinguishable from the old file format if the old file count happened to match.  I guess we could pick a magic number that is unlikely to be a file count, like something really big?

Reply:
So with the change that I PR'd I'd say the probability that someone has an archive that passes the sanity check is basically 0. My recommendation is that any future compressor change, we come up with a versioning mechanism for the archive. Probably just start with a magic string like you suggest that has no chance of ever naturally appearing